controllers/agents/app/serviceclaim_controller.go

```diff
@@ -43,11 +61,141 @@ type ServiceClaimReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
 func (r *ServiceClaimReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	l := log.FromContext(ctx).WithValues("namespace", req.Namespace, "name", req.Name)
+	l.Info("Reconciling service claim")
+
+	var sclaim primazaiov1alpha1.ServiceClaim
+	if err := r.Get(ctx, req.NamespacedName, &sclaim); err != nil {
+		l.Info("unable to retrieve ServiceClaim", "error", err)
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	config, remote_namespace, err := r.GetPrimazaKubeconfig(ctx, sclaim.Namespace)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	l.Info("remote cluster", "address", config.Host)
+
+	remote_client, err := client.New(config, client.Options{
+		Scheme: r.Client.Scheme(),
+		Mapper: r.Mapper,
+	})
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	objKey := client.ObjectKey{
+		Name:      constants.ApplicationAgentDeploymentName,
+		Namespace: req.NamespacedName.Namespace,
+	}
+	var deployment appsv1.Deployment
+	if err := r.Get(ctx, objKey, &deployment); err != nil {
+		l.Error(err, "unable to retrieve agent app deployment")
+		// we'll ignore not-found errors, since they can't be fixed by an immediate
+		// requeue (we'll need to wait for a new notification), and we can get them
+		// on deleted requests
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// examine DeletionTimestamp to determine if object is under deletion
+	if sclaim.DeletionTimestamp.IsZero() {
+		// The object is not being deleted, so if it does not have our finalizer,
+		// then lets add the finalizer and update the object. This is equivalent
+		// registering our finalizer.
+		if !controllerutil.ContainsFinalizer(&sclaim, ServiceClaimFinalizer) {
+			controllerutil.AddFinalizer(&sclaim, ServiceClaimFinalizer)
+			if err := r.Update(ctx, &sclaim); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+	} else {
+		// The object is being deleted
+		if controllerutil.ContainsFinalizer(&sclaim, ServiceClaimFinalizer) {
+			// our finalizer is present, so lets handle any external dependency
+			sclaimCopy := sclaim.DeepCopy()
+			sclaimCopy.Spec.EnvironmentTag = ""
+			sclaimCopy.Spec.ApplicationClusterContext = &primazaiov1alpha1.ServiceClaimApplicationClusterContext{}
+			sclaimCopy.Spec.ApplicationClusterContext.ClusterEnvironmentName = deployment.Labels["primaza.io/cluster-environment"]
+			sclaimCopy.Spec.ApplicationClusterContext.Namespace = sclaim.Namespace
+			sclaimCopy.Namespace = remote_namespace
+			sclaimCopy.ResourceVersion = ""
```

Comment:
Setting the ResourceVersion is not required. Both here and on L146

Reply:
I am getting an error when `ResourceVersion` is non-empty.